pkgs/development/tools/pnpm/fetch-deps/default.nix

```
@@ -38,11 +40,18 @@
           jq
           moreutils
           pnpm
+          yq
         ];
 
         installPhase = ''
           runHook preInstall
 
+          lockfileVersion="$(yq -r .lockfileVersion pnpm-lock.yaml)"
+          if [[ ''${lockfileVersion:0:1} -gt ${lib.versions.major pnpm.version} ]]; then
+            echo "ERROR: lockfileVersion in pnpm-lock.yaml is to new for the provided pnpm version ${lib.versions.major pnpm.version}!"
```

I started with some corrections and improvements, and continued with a suggested way to overcome this check.

```suggestion
            echo "ERROR: lockfileVersion $lockfileVersion in pnpm-lock.yaml is" \
              "too new for the provided pnpm version" \
              "${lib.versions.major pnpm.version}! Set" \
              "pnpmDontCheckLockfileVersion=true to avoid this error if you're" \
              "sure it won't cause other issues" >&2
```